File: Semester 4/Advanced Programming/RpcProtocolJava/ApplicationRepository/src/main/java/app/repository/RezervareDbRepo.java
package app.repository;

import app.model.Rezervare;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RezervareDbRepo implements IRezervareRepo {
    private DbUtils dbutils;
    private Logger logger = LogManager.getLogger();
    public RezervareDbRepo(Properties props){
        logger.info("Constructing RezervareDbRepo with properties : {}",props);
        dbutils=new DbUtils(props);
    }
    @Override
    public void save(Rezervare entity) {
        logger.traceEntry();
        Connection con = dbutils.getConnection();
        try(PreparedStatement stm = con.prepareStatement("insert into Rezervari (idCursa, nrLoc, client) values (?,?,?)")){
            stm.setInt(1,entity.getIdCursa());
            stm.setInt(2,entity.getNrLoc());
            stm.setString(3,entity.getClient());
            int result = stm.executeUpdate();
        }catch (SQLException ex){
            logger.error(ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Integer integer) {

    }

    @Override
    public void update(Rezervare entity) {

    }

    @Override
    public Rezervare findOne(Integer integer) {
        return null;
    }

    @Override
    public List<Rezervare> findAll() {
        return null;
    }

    @Override
    public List<Rezervare> findByIdCursa(Integer idCursa) {
        logger.traceEntry("Entered findByIdCursa: {}", idCursa);
        List<Rezervare> rezervari = new ArrayList<>();
        Connection con = dbutils.getConnection();
        try(PreparedStatement stm = con.prepareStatement("select * from Rezervari where idCursa=?")){
            stm.setInt(1,idCursa);
            try(ResultSet result = stm.executeQuery()){
                while(result.next()){
                    int id = result.getInt(1);
                    int idCursa2 = result.getInt(2);
                    int nrLoc = result.getInt(3);
                    String client = result.getString(4);
                    rezervari.add(new Rezervare(id,idCursa2,nrLoc,client));
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
        }
        logger.traceExit(rezervari);
        return rezervari;
    }
}

File: Semester 4/Advanced Programming/RpcProtocolC#/Repository/RezervareDbRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace Repositories
{
    public class RezervareDbRepo : IRezervareRepo
    {
        public RezervareDbRepo()
        {

        }
        public void Delete(int id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Rezervare> FindAll()
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Rezervare> FindByIdCursa(int idCursa)
        {
            IList<Rezervare> rezervari = new List<Rezervare>();
            var con = DBUtils.getConnection();
            using (var cmd = con.CreateCommand())
            {
                cmd.CommandText = "select * from Rezervari where idCursa=@idCursa";
                var param = cmd.CreateParameter();
                param.ParameterName = "@idCursa";
                param.Value = idCursa;
                cmd.Parameters.Add(param);
                using (var dataReader = cmd.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        int id = dataReader.GetInt32(0);
                        int idCursa2 = dataReader.GetInt32(1);
                        int nrLoc = dataReader.GetInt32(2);
                        string client = dataReader.GetString(3);
                        rezervari.Add(new Rezervare(id, idCursa2, nrLoc, client));
                    }
                }
            }
            con.Close();
            return rezervari;
        }

        public Rezervare FindOne(int id)
        {
            throw new NotImplementedException();

        }

        public void Save(Rezervare entity)
        {
            var con = DBUtils.getConnection();
            using (var cmd = con.CreateCommand())
            {
                cmd.CommandText = "insert into Rezervari (idCursa,nrLoc,client) values(@idCursa,@nrLoc,@client)";
                var paramId = cmd.CreateParameter();
                var paramNrLoc = cmd.CreateParameter();
                var paramClient = cmd.CreateParameter();

                paramId.ParameterName = "@idCursa";
                paramNrLoc.ParameterName = "@nrLoc";
                paramClient.ParameterName = "@client";

                paramId.Value = entity.idCursa;
                paramNrLoc.Value = entity.nrLoc;
                paramClient.Value = entity.client;

                cmd.Parameters.Add(paramId);
                cmd.Parameters.Add(paramNrLoc);
                cmd.Parameters.Add(paramClient);

                int result = cmd.ExecuteNonQuery();
            }
            con.Close();
        }

        public void Update(Rezervare entity)
        {
            throw new NotImplementedException();
        }
    }
}
